BraintreeUIKit/Models/BTUIKCardType.m

```diff
@@ -84,17 +84,9 @@ + (NSArray *)possibleCardTypesForNumber:(NSString *)number {
         return [[self class] allCards];
     }
     NSMutableSet *possibleCardTypes = [NSMutableSet set];
-    for (BTUIKCardType *cardType in [[self class] allCards]) {
-        for (NSString *prefix in cardType.validNumberPrefixes) {
-            NSUInteger compareLength = MIN(prefix.length, number.length);
-
-            NSString *sizedPrefix = [prefix substringToIndex:compareLength];
-            NSString *sizedNumber = [number substringToIndex:compareLength];
-            if ([sizedNumber isEqualToString:sizedPrefix]) {
-                [possibleCardTypes addObject:cardType];
-                break;
-            }
-        }
+    BTUIKCardType *card = [self cardTypeForNumber: number];
+    if (card) {
+        [possibleCardTypes addObject:card];
```

Comment:
Technically it should - yes. I think that we might want to just remove this method altogether. It is only used in one location in `BTUIKCardNumberFormField.m`:

```
- (BOOL)isPotentiallyValid {
    return [BTUIKCardType possibleCardTypesForNumber:self.number].count > 0;
}
```

So I just returned the first one since it would preserve the logic. Should we just rip it out and change the logic to:

```
- (BOOL)isPotentiallyValid {
    return [BTUIKCardType cardTypeForNumber:self.number] != nil;
}
```

I believe we need this method before to compare our hardcoded prefixes to see if if a short string (1-3 chars) COULD be valid. I think the new regex bypasses that.